handlers/sf-emails-to-s3-exporter/src/main/scala/com/gu/sf_emails_to_s3_exporter/SFConnector.scala

```diff
@@ -3,29 +3,29 @@ package com.gu.sf_emails_to_s3_exporter
 import com.gu.sf_emails_to_s3_exporter.ConfirmationWriteBackToSF.{EmailMessageToUpdate, EmailMessagesToUpdate}
 import com.gu.sf_emails_to_s3_exporter.Handler.safely
 import com.typesafe.scalalogging.LazyLogging
-import io.circe.Error
 import io.circe.generic.auto._
 import io.circe.parser.decode
 import io.circe.syntax.EncoderOps
+import io.circe.{Decoder, Error}
 import scalaj.http.{Http, HttpOptions}
 
 object SFConnector extends LazyLogging {
 
   case class SfAuthDetails(access_token: String, instance_url: String)
 
-  def getEmailsFromSfByQuery(sfAuthDetails: SfAuthDetails, sfApiVersion: String): Either[Error, EmailsFromSfResponse.Response] = {
-    logger.info("Getting emails from sf by query...")
+  def getRecordsFromSF[A: Decoder](sfAuthDetails: SfAuthDetails, sfApiVersion: String, query: String, batchSize: Integer): Either[Error, A] = {
```

Comment:
minor reduction in duplication, you would just create the SFConnector at the start of the function in the calling code, and then when you use it you needn't reference the common parameters.
```suggestion
class SFConnector(sfAuthDetails: SfAuthDetails, sfApiVersion: String) extends LazyLogging {

  case class SfAuthDetails(access_token: String, instance_url: String)
  
  def getRecordsFromSF[A: Decoder](query: String, batchSize: Integer): Either[Error, A] = {
```